### PR TITLE
Bug fixes in Sonata h5 parsing

### PR DESCRIPTION
**Pull request by hernando**
And other minor changes

### CODE REVIEW

File: brion/plugin/compartmentReportHDF5.cpp [hunk 1]
```diff
@@ -122,15 +113,25 @@ size_t _parseCacheSizeOption(const URI& uri)
...
+{
+    const auto keyValueIter = uri.findQuery("cache_size");
+    if (keyValueIter == uri.queryEnd())
+        return 0;
+
+    const auto& value = keyValueIter->second;
+    if (value == "auto")
+        return _autoCacheSize;
+
+    return _parseSizeOption(value, "cache_size");
```
Comment (tribal-tec) [inline brion/plugin/compartmentReportHDF5.cpp]: constants for the  magic strings here?

File: brion/plugin/compartmentReportHDF5.cpp [hunk 2]
```diff
@@ -189,9 +190,13 @@ std::string CompartmentReportHDF5::getDescription()
...
     return "SONATA HDF5 compartment reports:  "
            "[file://]/path/to/report.(h5|hdf5)"
-           "[?[cache_size=(auto|num_bytes):][cells_to_frames=(inf|ratio)]]\n"
-           "    default cache size is 3MB, auto will reserve space for a whole "
-           "frame or trace. The actual size depends on the chunk dimensions";
+           "[?[cache_size=(auto|num_bytes)&][cells_to_frames=(inf|ratio)&]"
+           "[chunk_size=bytes]]\n"
+           "    Byte counts can by suffixed by K or M.\n"
+           "    The cache is disabled by default, auto will reserve space for"
+           "a whole frame or trace, whatever is bigger. The actual size depends"
```
Comment (tribal-tec) [inline brion/plugin/compartmentReportHDF5.cpp]: space missing between 'for' and 'a'?

**Comment by hernando** [comment]
Retest this please

**Comment by hernando** [comment]
Retest this please

**Pull request merged**